### DISCUSSION TITLE
How can I read long string in DuckDB, or How can I export data from duckdb to an external file

### DISCUSSION THREAD

**Discussion by LuWei-HUST** [duckdb/duckdb#14924] [Q&A] [answered]
It‘s tricky to read long string in duckdb
try to check the tpch query:
![image](https://github.com/user-attachments/assets/e2bbf6e3-5b7f-4794-b7b1-e4e672aa69f0)

or
can i export it ?

**Comment by dmcmurchy** [comment]
@LuWei-HUST  - You should be able to output the results of the query to a file using [.output](https://duckdb.org/docs/api/cli/dot_commands#output-writing-results-to-a-file) command. Then you can open that file in your favourite editor.

**Reply by LuWei-HUST** [reply]
thanks, that helps

**Comment by soerenwolfers** [comment]
You can run the following in a Jupyter notebook after installing the Python client via `pip install duckdb`: 

```python
import duckdb
duckdb.query("""
FROM tpch_queries()
""").df().iloc[0, 1]

**Reply by LuWei-HUST** [reply]
thanks, that works too

**Comment by c-herrewijn** [comment, marked as answer]
You can switch output mode with "dot command" `.mode`.
With `.mode line` you get a nicely formatted result, [see docs for more options](https://duckdb.org/docs/api/cli/output_formats.html). You can use `.mode duckbox`  (or restart the CLI) to go back to the default.
```sql
.mode line
select query from tpch_queries() where query_nr = 1;
```

result:
```
query = SELECT
    l_returnflag,
    l_linestatus,
    sum(l_quantity) AS sum_qty,
    sum(l_extendedprice) AS sum_base_price,
    sum(l_extendedprice * (1 - l_discount)) AS sum_disc_price,
    sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) AS sum_charge,
    avg(l_quantity) AS avg_qty,
    avg(l_extendedprice) AS avg_price,
    avg(l_discount) AS avg_disc,
    count(*) AS count_order
FROM
    lineitem
WHERE
    l_shipdate <= CAST('1998-09-02' AS date)
GROUP BY
    l_returnflag,
    l_linestatus
ORDER BY
    l_returnflag,
    l_linestatus;
```

**Reply by LuWei-HUST** [reply]
thanks, that works
![image](https://github.com/user-attachments/assets/3bc2cded-14f5-40e3-8069-3d46ae580b4b)
